### PR TITLE
Move all artifacts (except logs) to cache dir 

### PR DESCRIPTION
**Pull request by tanin47**
Move everything except `~/.clusterfuzz/logs` to be under `~/.clusterfuzz/cache`, so that we can remove `~/.clusterfuzz/cache` on every CI run and keeps the logs folder.

Currently, the logs are deleted after every run on CI... so it's difficult to debug.

### CODE REVIEW

File: tool/tests/clusterfuzz/testcase_test.py [hunk 1]
```diff
@@ -106,15 +106,16 @@ def setUp(self): #pylint: disable=missing-docstring
         'clusterfuzz.testcase.Testcase.get_true_testcase_path'])
     self.mock.get_stored_auth_header.return_value = 'Bearer 1a2s3d4f'
     self.testcase_dir = os.path.expanduser(os.path.join(
-        '~', '.clusterfuzz', 'testcases', '12345_testcase'))
+        '~', '.clusterfuzz', 'cache', 'testcases', '12345_testcase'))
```
Comment (inferno-chromium) [inline tool/tests/clusterfuzz/testcase_test.py]: Can you use testcase.CLUSTERFUZZ_TESTCASES_DIR here as well ? Maybe a few more places need it.

File: tool/tests/clusterfuzz/testcase_test.py [hunk 2]
```diff
@@ -150,14 +151,15 @@ def test_zipfile(self):
     self.test.absolute_path = 'to/testcase.js'
     self.assertEqual(
         os.path.expanduser(
-            '~/.clusterfuzz/testcases/12345_testcase/to/testcase.js'),
+            '~/.clusterfuzz/cache/testcases/12345_testcase/to/testcase.js'),
```
Comment (inferno-chromium) [inline tool/tests/clusterfuzz/testcase_test.py]: Can you use testcase.CLUSTERFUZZ_TESTCASES_DIR here as well ?

**Comment by tanin47** [comment]
Coveralls seems to come back. I'll write the missing tests in the next PR because the tests aren't related to this PR.

File: tool/tests/clusterfuzz/testcase_test.py [hunk 3]
```diff
@@ -150,14 +151,15 @@ def test_zipfile(self):
...
         self.test.get_true_testcase_path('abcd.zip'))
 
     self.mock.ZipFile.assert_has_calls([
         mock.call(os.path.expanduser(
-            '~/.clusterfuzz/testcases/12345_testcase/abcd.zip'), 'r'),
-        mock.call().extractall(
-            os.path.expanduser('~/.clusterfuzz/testcases/12345_testcase'))])
+            '~/.clusterfuzz/cache/testcases/12345_testcase/abcd.zip'), 'r'),
```
Comment (inferno-chromium) [inline tool/tests/clusterfuzz/testcase_test.py]: Same as above.

File: tool/tests/clusterfuzz/testcase_test.py (hunk 1 above)
Comment (tanin47) [inline tool/tests/clusterfuzz/testcase_test.py]: Done. Thanks.

File: tool/tests/clusterfuzz/testcase_test.py (hunk 3 above)
Comment (tanin47) [inline tool/tests/clusterfuzz/testcase_test.py]: Done. Thanks.

**Pull request merged**